libsolidity/parsing/Parser.cpp

```diff
@@ -1986,23 +1985,34 @@ pair<vector<ASTPointer<Expression>>, vector<ASTPointer<ASTString>>> Parser::pars
 	return ret;
 }
 
-optional<string> Parser::findLicenseString()
+optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> const& _nodes)
 {
 	// We circumvent the scanner here, because it skips non-docstring comments.
-	static regex const licenseRegex("SPDX-License-Identifier:\\s+([^\\n\\r]+)");
-
+	static regex const licenseRegex("SPDX-License-Identifier:\\s*([^\\n\\r\\s]+)");
+
+	// Remove all parts of the source that where referenced by different AST node locations.
```

Comment:
That's a good idea, but we could also find all places that match the regex and see if we ended up inside one of the nodes.